common/smoke-test/.gitignore

```diff
@@ -1,4 +0,0 @@
-!package-lock.json
```

Comment:
Is it safe to remove this .gitignore?  Why were these rules required previously, and why are they no longer required?  The `!package-lock.json` is particularly interesting since it was explicitly overriding the base .gitignore.

Reply:
It's a good question (and I appreciate you wanting to be careful about making changes without understanding the original context). In this case, it looks like the .gitignore was a holdover from the previous smoke test model that did not use auto-generated files (https://github.com/Azure/azure-sdk-for-js/pull/5072/files). Since the package.json file gets automatically updated in the dev loop, there can't be a repeatable build definition in git via the lock file, and indeed the lock file wasn't even checked in anymore.

That said, we COULD check in the generated lock file that the smoke test pipeline creates when running all the smoke tests, but I'm not sure whether that's worth it.